src/main/scala/cognite/spark/v1/RawJsonConverter.scala

```diff
@@ -0,0 +1,354 @@
+      case v: Row => rowToJson(v)
+      case _ =>
+        throw new CdfSparkIllegalArgumentException(
+          s"Value $v of type ${v.getClass} can not be written to RAW")
+    }
+
+  def rowToJson(r: Row): Json =
+    if (r.schema != null) {
+      Json.obj(r.schema.fieldNames.map(f => f -> anyToRawJson(r.getAs[Any](f))): _*)
+    } else {
+      Json.arr(r.toSeq.map(anyToRawJson): _*)
+    }
+
+  def rowsToRawItems(
+      nonKeyColumnNames: Seq[String],
+      keyColumnName: String,
+      rows: Seq[Row]): Seq[RawRow] =
+    rows.map { row =>
+      val key = row.getString(row.fieldIndex(keyColumnName))
+      if (key == null) {
+        throw new CdfSparkIllegalArgumentException("\"key\" can not be null.")
+      }
+      RawRow(
+        key,
+        nonKeyColumnNames.map(f => f -> anyToRawJson(row.getAs[Any](f))).toMap
+      )
+    }
+
+  case class RawJsonColumn(jsonName: String, columnName: String, dataType: DataType)
+
+  def untypedRowConverter(row: RawRow): Row =
+    new GenericRowWithSchema(
+      Array(
+        row.key,
+        row.lastUpdatedTime.map(java.sql.Timestamp.from).orNull,
+        JsonObject(row.columns.toSeq: _*).asJson.noSpaces
+      ),
+      RawTableRelation.defaultSchema
+    )
+
+  def makeRowConverter(
+      sparkSchema: StructType,
+      jsonFieldNames: Array[String],
+      lastUpdatedTimeColumn: String,
+      keyColumn: String): RawRow => Row = {
+    assert(sparkSchema.fieldNames(0) == keyColumn, "Key must be the first column")
+    assert(
+      sparkSchema.fieldNames(1) == lastUpdatedTimeColumn,
+      "LastUpdatedTime must be the second column")
+
+    val fieldConverters = sparkSchema.fields.drop(2).map(f => makeConverter(f.dataType))
+    val nameMap = new java.util.HashMap[String, Int]()
+    for ((field, i) <- jsonFieldNames.zipWithIndex) {
+      nameMap.put(field, i)
+    }
+
+    row =>
+      {
+        val rowArray = new Array[Any](sparkSchema.fields.length)
+        rowArray(0) = row.key
+        rowArray(1) = row.lastUpdatedTime.map(java.sql.Timestamp.from).orNull
+        for ((key, value) <- row.columns) {
+          val fieldIndex = nameMap.getOrDefault(key, -1)
+          if (fieldIndex >= 0) {
+            rowArray(fieldIndex + 2) = fieldConverters(fieldIndex).convertNullSafe(value)
+          }
+        }
+        new GenericRowWithSchema(rowArray, sparkSchema)
+      }
+  }
+
+  // private lazy val timestampFormatter = TimestampFormatter(
+  //   None,
+  //   java.time.ZoneOffset.UTC,
+  //   Locale.US,
+  //   legacyFormat = FAST_DATE_FORMAT,
+  //   isParsing = true)
+  // private lazy val timestampNTZFormatter = TimestampFormatter(
+  //   options.timestampFormatInRead,
+  //   options.zoneId,
+  //   legacyFormat = FAST_DATE_FORMAT,
+  //   isParsing = true,
+  //   forTimestampNTZ = true)
+  // private lazy val dateFormatter = DateFormatter(
+  //   options.dateFormatInRead,
+  //   options.locale,
+  //   legacyFormat = FAST_DATE_FORMAT,
+  //   isParsing = true)
```

Comment:
I wanted to leave it here, since I didn't add support for parsing timestamps, but we might want to do it later. This is the code Spark uses for that, which I used as a reference to implement the other types, so I thought leaving this as a reference for the rest.